src/dataset.cpp

```diff
@@ -435,17 +472,40 @@ DatasetSlice DatasetSlice::operator-=(const ConstDatasetSlice &other) {
       [](VariableSlice &a, const ConstVariableSlice &b) { return a -= b; },
       *this, other);
 }
+DatasetSlice DatasetSlice::operator-=(const double value) {
+  for (auto var : *this)
+    if (var.tag() == Data::Value{})
+      var -= value;
+  return *this;
+}
 
 DatasetSlice DatasetSlice::operator*=(const Dataset &other) {
   return times_equals(*this, other);
 }
 DatasetSlice DatasetSlice::operator*=(const ConstDatasetSlice &other) {
   return times_equals(*this, other);
 }
+DatasetSlice DatasetSlice::operator*=(const double value) {
+  for (auto var : *this)
+    if (var.tag() == Data::Value{})
+      var *= value;
+    else if (var.tag() == Data::Variance{})
+      var *= value * value;
+  return *this;
+}
 
 Dataset operator+(Dataset a, const Dataset &b) { return a += b; }
 Dataset operator-(Dataset a, const Dataset &b) { return a -= b; }
 Dataset operator*(Dataset a, const Dataset &b) { return a *= b; }
+Dataset operator+(Dataset a, const ConstDatasetSlice &b) { return a += b; }
```

Comment:
Why the operator+ affects the first argument? It seems that it should be somehow:
`Dataset operator+(const Dataset& a, const ConstDatasetSlice &b) { auto copy(a); return copy += b; }`
The same question for `Variable`.

Reply:
It is a simple way of supporting both lvalue and rvalue arguments. This way you can do

```cpp
dataset = std::move(dataset) + x;
```
without making any copies or allocating new memory.

The alternative would be to provide overloads for `const Dataset &` and `Dataset &&`. That would give more options when handling exceptions, but for now passing by value is briefer and sufficient, I think.